nntrainer/graph/network_graph.cpp

```diff
@@ -414,45 +412,34 @@ void NetworkGraph::setOutputLayers() {
     }
 
     if (layer_idx->getNumOutputs() != layer_idx->output_layers.size()) {
-      layer_idx->setNumOutputs(layer_idx->output_layers.size());
+      if (layer_idx->output_layers.size() == 0) {
+        /** No output layer inplies its the last layer */
+        layer_idx->setNumOutputs(1);
+        layer_idx->output_layers.clear();
+        layer_idx->output_layers.push_back("__exit__");
+        last_layer_count += 1;
+      } else if (layer_idx->getNumOutputs() < layer_idx->output_layers.size()) {
+        /** this if the multi-output layer */
```

Comment:
is `layer_idx->getNumOutputs() < `layer_idx->output_layers.size()` holds for multi-output layer? I wonder as realizeMultiOutput is setting the output.

Reply:
Yes. `RealizeMultiOutput` has a different job. It is to insert an `outputlayer` and set this correctly. However, if an `outputlayer` is already set, then its `numOutputs` and `output_layers.size()` don't match.

Reply:
Understood 👍